diff -urN libmpeg3-1.2.3.org/audio/mpeg3audio.h libmpeg3-1.2.3/audio/mpeg3audio.h
--- libmpeg3-1.2.3.org/audio/mpeg3audio.h	Sat Feb 10 22:55:52 2001
+++ libmpeg3-1.2.3/audio/mpeg3audio.h	Sat Feb 10 22:56:44 2001
@@ -1,8 +1,8 @@
 #ifndef MPEG3AUDIO_H
 #define MPEG3AUDIO_H
 
-#include "ac3.h"
-#include "../bitstream.h"
+#include <libmpeg3/audio/ac3.h>
+#include <libmpeg3/bitstream.h>
 
 #define MAXFRAMESIZE 1792
 #define HDRCMPMASK 0xfffffd00
diff -urN libmpeg3-1.2.3.org/bitstream.h libmpeg3-1.2.3/bitstream.h
--- libmpeg3-1.2.3.org/bitstream.h	Sat Feb 10 22:55:52 2001
+++ libmpeg3-1.2.3/bitstream.h	Sat Feb 10 22:57:06 2001
@@ -1,7 +1,7 @@
 #ifndef BITSTREAM_H
 #define BITSTREAM_H
 
-#include "mpeg3demux.h"
+#include <libmpeg3/mpeg3demux.h>
 #include <sys/types.h>
 
 //                                    next bit in forward direction
diff -urN libmpeg3-1.2.3.org/libmpeg3.h libmpeg3-1.2.3/libmpeg3.h
--- libmpeg3-1.2.3.org/libmpeg3.h	Sat Feb 10 22:55:52 2001
+++ libmpeg3-1.2.3/libmpeg3.h	Sat Feb 10 22:57:33 2001
@@ -5,7 +5,7 @@
 extern "C" {
 #endif
 
-#include "mpeg3private.h"
+#include <libmpeg3/mpeg3private.h>
 
 
 /* Supported color models for mpeg3_read_frame */
diff -urN libmpeg3-1.2.3.org/mpeg3css.h libmpeg3-1.2.3/mpeg3css.h
--- libmpeg3-1.2.3.org/mpeg3css.h	Sat Feb 10 22:55:52 2001
+++ libmpeg3-1.2.3/mpeg3css.h	Sat Feb 10 22:57:57 2001
@@ -2,7 +2,7 @@
 #define MPEG3CSS_H
 
 
-#include "mpeg3private.inc"
+#include <libmpeg3/mpeg3private.inc>
 
 struct mpeg3_block 
 {
diff -urN libmpeg3-1.2.3.org/mpeg3demux.h libmpeg3-1.2.3/mpeg3demux.h
--- libmpeg3-1.2.3.org/mpeg3demux.h	Sat Feb 10 22:55:52 2001
+++ libmpeg3-1.2.3/mpeg3demux.h	Sat Feb 10 22:58:17 2001
@@ -1,7 +1,7 @@
 #ifndef MPEG3DEMUX_H
 #define MPEG3DEMUX_H
 
-#include "mpeg3title.h"
+#include <libmpeg3/mpeg3title.h>
 #include <stdio.h>
 
 typedef struct
diff -urN libmpeg3-1.2.3.org/mpeg3io.h libmpeg3-1.2.3/mpeg3io.h
--- libmpeg3-1.2.3.org/mpeg3io.h	Sat Feb 10 22:55:52 2001
+++ libmpeg3-1.2.3/mpeg3io.h	Sat Feb 10 22:58:38 2001
@@ -3,8 +3,8 @@
 
 
 #include <stdio.h>
-#include "mpeg3css.h"
-#include "mpeg3private.inc"
+#include <libmpeg3/mpeg3css.h>
+#include <libmpeg3/mpeg3private.inc>
 
 /* Filesystem structure */
 
diff -urN libmpeg3-1.2.3.org/mpeg3private.h libmpeg3-1.2.3/mpeg3private.h
--- libmpeg3-1.2.3.org/mpeg3private.h	Sat Feb 10 22:55:52 2001
+++ libmpeg3-1.2.3/mpeg3private.h	Sat Feb 10 22:59:34 2001
@@ -1,12 +1,12 @@
 #ifndef MPEG3PRIVATE_H
 #define MPEG3PRIVATE_H
 
-#include "mpeg3atrack.h"
-#include "mpeg3css.h"
-#include "mpeg3io.h"
-#include "mpeg3private.inc"
-#include "mpeg3title.h"
-#include "mpeg3vtrack.h"
+#include <libmpeg3/mpeg3atrack.h>
+#include <libmpeg3/mpeg3css.h>
+#include <libmpeg3/mpeg3io.h>
+#include <libmpeg3/mpeg3private.inc>
+#include <libmpeg3/mpeg3title.h>
+#include <libmpeg3/mpeg3vtrack.h>
 
 typedef struct
 {
diff -urN libmpeg3-1.2.3.org/mpeg3title.h libmpeg3-1.2.3/mpeg3title.h
--- libmpeg3-1.2.3.org/mpeg3title.h	Sat Feb 10 22:55:52 2001
+++ libmpeg3-1.2.3/mpeg3title.h	Sat Feb 10 22:59:48 2001
@@ -1,7 +1,7 @@
 #ifndef MPEG3TITLE_H
 #define MPEG3TITLE_H
 
-#include "mpeg3io.h"
+#include <libmpeg3/mpeg3io.h>
 
 typedef struct
 {
diff -urN libmpeg3-1.2.3.org/mpeg3vtrack.h libmpeg3-1.2.3/mpeg3vtrack.h
--- libmpeg3-1.2.3.org/mpeg3vtrack.h	Sat Feb 10 22:55:52 2001
+++ libmpeg3-1.2.3/mpeg3vtrack.h	Sat Feb 10 23:00:11 2001
@@ -1,8 +1,8 @@
 #ifndef MPEG3_VTRACK_H
 #define MPEG3_VTRACK_H
 
-#include "mpeg3demux.h"
-#include "video/mpeg3video.h"
+#include <libmpeg3/mpeg3demux.h>
+#include <libmpeg3/video/mpeg3video.h>
 
 typedef struct
 {
diff -urN libmpeg3-1.2.3.org/video/mpeg3video.h libmpeg3-1.2.3/video/mpeg3video.h
--- libmpeg3-1.2.3.org/video/mpeg3video.h	Sat Feb 10 22:55:52 2001
+++ libmpeg3-1.2.3/video/mpeg3video.h	Sat Feb 10 23:01:13 2001
@@ -1,11 +1,11 @@
 #ifndef MPEGVIDEO_H
 #define MPEGVIDEO_H
 
-#include "../bitstream.h"
-#include "../mpeg3private.inc"
-#include "idct.h"
-#include "slice.h"
-#include "../timecode.h"
+#include <libmpeg3/bitstream.h>
+#include <libmpeg3/mpeg3private.inc>
+#include <libmpeg3/video/idct.h>
+#include <libmpeg3/video/slice.h>
+#include <libmpeg3/timecode.h>
 
 /* zig-zag scan */
 extern unsigned char mpeg3_zig_zag_scan_nommx[64];
