--- libmpeg3-1.5.1/audio/mpeg3audio.c.orig	Wed May 28 23:33:05 2003
+++ libmpeg3-1.5.1/audio/mpeg3audio.c	Wed May 28 23:33:13 2003
@@ -532,9 +532,9 @@
 			index = audio->sample_seek / MPEG3_AUDIO_CHUNKSIZE;
 			if(index >= track->total_sample_offsets) index = track->total_sample_offsets - 1;
 			title_number = (track->sample_offsets[index] & 
-				0xff00000000000000) >> 56;
+				0xff00000000000000LL) >> 56;
 			byte = track->sample_offsets[index] &
-				0xffffffffffffff;
+				0xffffffffffffffLL;
 
 			mpeg3demux_open_title(demuxer, title_number);
 			mpeg3demux_seek_byte(demuxer, byte);
--- libmpeg3-1.5.1/video/seek.c.orig	Wed May 28 23:32:47 2003
+++ libmpeg3-1.5.1/video/seek.c	Wed May 28 23:33:00 2003
@@ -263,9 +263,9 @@
 
 						frame = track->keyframe_numbers[i];
 						title_number = (track->frame_offsets[frame] & 
-							0xff00000000000000) >> 56;
+							0xff00000000000000LL) >> 56;
 						byte = track->frame_offsets[frame] & 
-							  0xffffffffffffff;
+							  0xffffffffffffffLL;
 
 						video->framenum = track->keyframe_numbers[i];
 
